File: CS/ReportAtRuntimeMvcApp/Controllers/HomeController.cs
using System;
using System.Linq;
using System.Web;
using System.Web.Mvc;

namespace ReportAtRuntimeMvcApp.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            return View();
        }

        public ActionResult Designer() {
            Models.ReportDesignerModel model = new Models.ReportDesignerModel();
            return View(model);
        }

        public ActionResult Viewer() {
            return View();
        }
    }
}

File: VB/ReportAtRuntimeMvcApp/Controllers/HomeController.vb
Imports System.Web.Mvc

Namespace ReportAtRuntimeMvcApp.Controllers

    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function Designer() As ActionResult
            Dim model As Models.ReportDesignerModel = New Models.ReportDesignerModel()
            Return View(model)
        End Function

        Public Function Viewer() As ActionResult
            Return View()
        End Function
    End Class
End Namespace
